pkg/vcdsdk/gateway.go

```diff
@@ -1949,3 +1950,40 @@ func (gm *GatewayManager) UpdateLoadBalancer(ctx context.Context, lbPoolName str
 	}
 	return vsSummary.VirtualIpAddress, nil
 }
+
+func (gm *GatewayManager) FetchIpSpacesBackingGateway(ctx context.Context) ([]string, error) {
+	ipSpaceService := gm.Client.APIClient.IpSpacesApi
+
+	filterString := fmt.Sprintf("gatewayId==%s", gm.GatewayRef.Id)
+	options := swaggerClient.IpSpacesApiGetFloatingIpSuggestionsOpts{Filter: optional.NewString(filterString), SortAsc: optional.NewString("ipSpaceRef.name"), SortDesc: optional.EmptyString()}
+	var pageSize int32 = 10
+	var pageNum int32 = 1
+	var resultTotal int32 = math.MaxInt32
+
+	var ipSpaceIds []string
+	for int32(math.Ceil(float64(resultTotal)/float64(pageSize))) >= pageNum {
+		suggestions, _, err := ipSpaceService.GetFloatingIpSuggestions(ctx, pageNum, pageSize, &options)
+		if err != nil {
+			return nil, err
+		}
+		resultTotal = suggestions.ResultTotal
+
+		if suggestions.Values != nil {
+			if len(suggestions.Values) == 0 {
+				break
+			}
+			for _, element := range suggestions.Values {
+				ipSpaceRef := element.IpSpaceRef
+				if ipSpaceRef != nil {
+					ipSpaceIds = append(ipSpaceIds, ipSpaceRef.Id)
+				}
+			}
+		} else {
+			return nil, fmt.Errorf("invalid values in page %d (pageSize %d) of GetFloatingIpSuggestions call for gateway %s", pageNum, pageSize, gm.NetworkName)
```

Comment:
Nit: TBH, we could just log and break here. We are establishing a contract (that the number of entries will remain constant while the loop runs), while VCD does not have any such contract. We could make this loop of collecting ipSpaceIds more 'lenient'.